### ISSUE TITLE
GCE node deletion times out - allow an async option

### ISSUE THREAD

**Issue by zimventures** [apache/libcloud#1356]
## Feature Request
Currently the GCE `node_delete()` function uses the `async_request()` call (which ironically isn't actually async). Deleting a basic n1-standard-1 node running Centos in us-central times out every time. This isn't an uncommon occurrence. Searching the intertubes there are other references of other projects having the same issue:

https://github.com/kubevirt/cloud-image-builder/issues/72
https://github.com/ansible/ansible/issues/29597

The current way to avoid the timeout is to use the `ex_destroy_multiple_nodes()` function with a gigantic timeout value. An option, for sure, but not flexible enough. 

An easy solution in this case would be to offer both a sync and async option for deletion. Applications can poll, if desired. Thankfully, the async option is fairly straightforward, defaulting to the current sync mode of operation:

```python
        if sync:
            self.connection.async_request(request, method='DELETE')
        else:
            self.connection.request(request, method='DELETE')
```

**Comment by Kami** [comment]
I think that's a reasonable approach although it would be good to have a consistent API for all the operations which are async (aka involve polling).

So ideally all of those methods would take ``sync=False`` / ``ex_sync=False`` argument.

**Comment by Kami** [comment]
Closing since this has been implemented as part of #1357.
